src/xtgeo/grid3d/_grdecl_format.py

```diff
@@ -0,0 +1,247 @@
+
+
+def interpret_number(numb):
+    """
+    Interpret a eclipse token as a number, ie. if it is not
+    an integer, try interpreting it as a float. raises
+    ValueError if not a number
+
+    >>> interpret_number("3")
+    3
+    >>> interpret_number("1.0")
+    1.0
+    >>> interpret_number("1.0E+30")
+    1e+30
+
+    """
+    try:
+        return int(numb)
+    except ValueError:
+        return float(numb)
+
+
+def interpret_token(val):
+    """
+    Interpret a eclipse token, tries to interpret the
+    value in the following order:
+    * string literal
+    * keyword
+    * repreated keyword
+    * number
+
+    If the token cannot be matched, we default to returning
+    the uninterpreted token.
+
+    >>> interpret_token("3")
+    [3]
+    >>> interpret_token("1.0")
+    [1.0]
+    >>> interpret_token("'hello'")
+    ['hello']
+    >>> interpret_token("PORO")
+    ['PORO']
+    >>> interpret_token("3PORO")
+    ['3PORO']
+    >>> interpret_token("3PORO")
+    ['3PORO']
+    >>> interpret_token("3*PORO")
+    ['PORO', 'PORO', 'PORO']
+    >>> interpret_token("3*'PORO '")
+    ['PORO ', 'PORO ', 'PORO ']
+    >>> interpret_token("3'PORO '")
+    ["3'PORO '"]
+
+    """
+    if val[0] == "'" and val[-1] == "'":
+        # A string literal
+        return [val[1:-1]]
+    if val[0].isalpha():
+        # A keyword
+        return [val]
+    if "*" in val:
+        multiplicand, value = val.split("*")
+        return interpret_token(value) * int(multiplicand)
+    try:
+        return [interpret_number(val)]
+    except ValueError:
+        return [val]
+
+
+IGNORE_ALL = None
+
+
+@contextmanager
+def open_grdecl(grdecl_file, keywords, max_len=None, ignore=IGNORE_ALL):
+    """
+    Opens the grdecl_file for reading. Is a generator for tuples of keyword /
+    values in records of that file. The format of the file must be that of the
+    GRID section of a eclipse input DATA file.
+
+    The records looked for must be "simple" ie.  start with the keyword, be
+    followed by single word values and ended by a slash ('/').
+
+    .. code-block:: none
+
+        KEYWORD
+        value value value /
+
+    reading the above file with :code:`open_grdecl("filename.grdecl",
+    keywords="KEYWORD")` will generate :code:`[("KEYWORD", ["value", "value",
+    "value"])]`
+
+    open_grdecl does not follow includes, obey skips, parse MESSAGE commands or
+    make exception for groups and subrecords.
+
+    Note: trailing spaces of keywords are ignored
+
+    Raises:
+        ValueError: when end of file is reached without terminating a keyword,
+            or the file contains an unrecognized (or ignored) keyword.
+
+    Args:
+        keywords (List[str]): Which keywords to look for, these are expected to
+        be at the start of a line in the file  and the respective values
+        following on subsequent lines separated by whitespace. Reading of a
+        keyword is completed by a final '\'. See example above.
```

Comment:
Although not a public docstring, it could be nice to follow the google docstring style:

https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

"""One line of summary, ending with punctuation.

Then after space, some text, Args etc...

"""
For Args, indentation is missing?

Reply:
Fixed now :)